common/setups/returnn_common/nnet_constructor.py

```diff
@@ -0,0 +1,425 @@
+        """
+        super().__init__()
+        self.extern_data = extern_data
+
+    def serialize_data(self, constructor_data: DataInitArgs) -> str:
+        """
+        Serialize a single DataInitArgs object
+
+        :param constructor_data: init args for serialization
+        """
+        content = ""
+        wrapped_dim_names = []
+        for c_dim in constructor_data.dim_tags:
+            wrapped_dim_names.append(CodeWrapper(c_dim.name))
+
+        if constructor_data.has_batch_dim:
+            wrapped_dim_names = [CodeWrapper("nn.batch_dim")] + wrapped_dim_names
+
+        if constructor_data.sparse_dim is not None:
+            sparse_dim = constructor_data.sparse_dim
+            sparse_dim_name = sparse_dim.name
+        else:
+            sparse_dim_name = None
+
+        content += textwrap.dedent(
+            f"""\
+        {constructor_data.name} = nn.Data(
+            name="{constructor_data.name}",
+            available_for_inference={constructor_data.available_for_inference},
+            dim_tags={wrapped_dim_names},
+            sparse_dim={sparse_dim_name},
+        )
+        """
+        )
+
+        return content
+
+    def get(self) -> str:
+        content = ""
+
+        # collect dims into a set to only write each Dim once if shared
+        dims = set()
+        for constructor_data in self.extern_data:
+            for dim in constructor_data.dim_tags:
+                dims.add(dim)
+            if constructor_data.sparse_dim:
+                dims.add(constructor_data.sparse_dim)
+
+        for dim in dims:
+            content += (
+                f"{dim.name} = nn.{'FeatureDim' if dim.is_feature else 'SpatialDim'}(\"{dim.name}\", "
+                f"{instanciate_delayed(dim.dim)})\n"
+            )
+
+        for constructor_data in self.extern_data:
+            content += self.serialize_data(constructor_data)
+
+        # RETURNN does not allow for "name" in the args, as this is set via the dict key
+        # thus, we need to explicitly remove it for now
+        for constructor_data in self.extern_data:
+            content += (
+                f"{constructor_data.name}_args = {constructor_data.name}.get_kwargs()\n"
+            )
+            content += f'{constructor_data.name}_args.pop("name")\n'
+
+        content += "\nextern_data={\n"
+        for constructor_data in self.extern_data:
+            content += f'    "{constructor_data.name}": {constructor_data.name}_args,\n'
+        content += "}\n"
+        return content
+
+
+class ReturnnCommonImport(SerializerObject):
+    """
+    A class to indicate a module or function that should be imported within the returnn config
+
+    When passed to the ReturnnCommonSerializer it will automatically detect the local package in case of
+    `make_local_package_copy=True`, unless specific package paths are given.
+    """
+
+    def __init__(
+        self,
+        code_object: str,
```

Comment:
Is this a string and not `tk.Path` because of the manipulations done in the init? Or does this have a different reason? If so maybe we should note somewhere that this format is not the same `tk.Path.get()` returns, so there is no confusion.

Reply:
Look at the example, this is an import definition and not a physical file or directory, so it can and should not be a tk.Path